javascript-carousel/main.js

```diff
@@ -0,0 +1,77 @@
+var i = 1;
+var prevI = null;
```

Comment:
I'm not sure i understand what this `prevI` variable is doing for you. You should be able to just use `i` to track the "index" of your image url as well as the index of the circle that is currently active.